File: ios/RnMercadopagoServices.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(RnMercadopagoServices, NSObject)

RCT_EXTERN_METHOD(getDevice: (RCTPromiseResolveBlock)resolve
                  rejecter: (RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/RnMercadopagoServices.swift
import MercadoPagoDevicesSDK

@objc(RnMercadopagoServices)
class RnMercadopagoServices: NSObject {

  @objc(getDevice:rejecter:)
  func getDevice(_ resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) -> Void {
    MercadoPagoDevicesSDK.shared.execute()
    let json = try MercadoPagoDevicesSDK.shared.getInfoAsJsonString()
    resolve(json)
  }
}
